dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/CompanyDataControllerTest.kt

```diff
@@ -200,4 +202,26 @@ class CompanyDataControllerTest {
             "The exception message does not say that a 403 client error was the cause.",
         )
     }
+
+    @Test
+    fun `post a dummy company twice and receive the expected error code`() {
+        val testCompanyInformation = apiAccessor.testDataProviderForEuTaxonomyDataForNonFinancials
+            .getCompanyInformationWithoutIdentifiers(1).first()
+        val randomIsin = CompanyIdentifier(
+            identifierValue = UUID.randomUUID().toString(),
+            identifierType = CompanyIdentifier.IdentifierType.isin,
+        )
+        val randomizedCompanyInformation = testCompanyInformation.copy(identifiers = listOf(randomIsin))
+        apiAccessor.tokenHandler.obtainTokenForUserType(UserType.Uploader)
+        apiAccessor.companyDataControllerApi.postCompany(randomizedCompanyInformation)
+        val exception =
+            assertThrows<Exception> {
+                apiAccessor.companyDataControllerApi.postCompany(randomizedCompanyInformation)
+            }
+        assertEquals(
+            "Client error : 400 ",
+            exception.message,
+            "The exception message does not say that a 400 client error was the cause.",
```

Comment:
I would have expected a check for the content "Could not insert company as one company identifier is already used to identify another company"

Reply:
This conflict with main has been addressed.

Reply:
Changed the test to directly work with the response instead of the thrown exception